sdk/servicebus/azure-messaging-servicebus-track2-perf/src/main/java/com/azure/messaging/servicebus/perf/ReceiveAndDeleteMessageTest.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus.perf;
+
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.ServiceBusMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import com.azure.messaging.servicebus.models.ReceiveMode;
+import com.azure.messaging.servicebus.perf.core.ServiceBusStressOptions;
+import com.azure.messaging.servicebus.perf.core.ServiceTest;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Performance test.
+ */
+public class ReceiveAndDeleteMessageTest extends ServiceTest<ServiceBusStressOptions> {
+    private final ClientLogger logger = new ClientLogger(ReceiveAndDeleteMessageTest.class);
+    private final ServiceBusStressOptions options;
+
+    /**
+     * Creates test object
+     * @param options to set performance test options.
+     */
+    public ReceiveAndDeleteMessageTest(ServiceBusStressOptions options) {
+        super(options, ReceiveMode.RECEIVE_AND_DELETE);
+        this.options = options;
+    }
+
+    @Override
+    public Mono<Void> globalSetupAsync() {
+        // Since test does warm up and test many times, we are sending many messages, so we will have them available.
+        return Mono.defer(() -> {
+            int total = options.getParallel() * options.getMessagesToSend() * TOTAL_MESSAGE_MULTIPLIER;
+            List<ServiceBusMessage> messages = new ArrayList<>();
+            for (int i = 0; i< total; ++i) {
+                ServiceBusMessage message =  new ServiceBusMessage(CONTENTS.getBytes(Charset.defaultCharset()));
+                message.setMessageId(UUID.randomUUID().toString());
+                messages.add(message);
+            }
+            return senderAsync.sendMessages(messages);
+        });
+    }
+
+    @Override
+    public void run() {
+        IterableStream<ServiceBusReceivedMessageContext> messages = receiver
+            .receiveMessages(options.getMessagesToReceive());
+        if (messages.stream().count() <= 0) {
+            throw logger.logExceptionAsWarning(new RuntimeException("Error. Should have received some messages."));
+        }
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        return receiverAsync
+            .receiveMessages()
+            .map(messageContext -> {
+                return messageContext;
```

Comment:
Why are we not checking the received msg count being > 0 here like other places ?